### ISSUE TITLE
[some question]  some question about `build_save_area()`

### ISSUE THREAD

**Issue by JustPlay** [IBM/sev-snp-measure#2]
in `build_save_area()`
1. the py code use `cr0=0x10`, but based-on https://www.amd.com/system/files/TechDocs/24593.pdf (page 103, 526), it said `cr0=6000_0010h `,  why ?

**Comment by dubek** [comment]
Good questions. Some of this is from empirical attempt of launching an SEV-ES guest and looking at the VMSA that KVM sees before encrypting it.

I'll look into this and add better documentation to the code.

**Comment by dubek** [comment]
For cr0, AMD APM Vol 2 Table 14-1 (page 526 in PDF) says:

> Value after INIT:
> CD and NW are unchanged
> Bit 4 (reserved) = 1
> All others = 0

So if CD and NW are 0, and bit 4 is 1, you get `cr0=0x10`.

**Comment by dubek** [comment]
Value of `g_pat` is taken from [this line](https://gitlab.com/qemu-project/qemu/-/blob/master/target/i386/cpu.c#L5944) in the QEMU initialization.

**Comment by JustPlay** [comment]
> For cr0, AMD APM Vol 2 Table 14-1 (page 526 in PDF) says:
> 
> > Value after INIT:
> > CD and NW are unchanged
> > Bit 4 (reserved) = 1
> > All others = 0
> 
> So if CD and NW are 0, and bit 4 is 1, you get `cr0=0x10`.

should we use the `INIT` status, not `RESET` ？， why?

**Comment by dubek** [comment]
> should we use the `INIT` status, not `RESET` ？， why?
> and，what's more， in amd‘s doc, it saids "CD and NW are unchanged", so how do you know the initial value? (because in RESET, CD=NW=1, not 0)

You're right, I'm not sure. But empirically this is the case (this is the CR0 value in the VMSA) when starting SEV-ES guests with QEMU and KVM. Maybe the documentation or git history there give some explanations for these choice. Let me know if you find anything.

**Comment by tlendacky** [comment]
> Value of `g_pat` is taken from [this line](https://gitlab.com/qemu-project/qemu/-/blob/master/target/i386/cpu.c#L5944) in the QEMU initialization. It also appears in kvm's code as `MSR_IA32_CR_PAT_DEFAULT`.

This is the PAT MSR reset value as documented in APM Volume 2, Section A.3.
